src/Rule/ContainsLowercaseCharacters.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Morebec\Validator\Rule;
+
+
+use InvalidArgumentException;
+use Morebec\Validator\ValidationRuleInterface;
+
+class ContainsLowercaseCharacters implements ValidationRuleInterface
+{
+    /**
+     * @var int
+     */
+    private $numberCharacters;
+
+    /**
+     * @var bool
+     */
+    private $strict;
+
+    /**
+     * @var string|null
+     */
+    private $message;
+
+    /**
+     * ContainsLowercaseCharacters constructor.
+     * @param int $numberCharacters
+     * @param bool $strict
+     * @param string|null $message
+     */
+    public function __construct(
+        int $numberCharacters,
+        bool $strict,
+        ?string $message = null
+    )
+    {
+        if($numberCharacters<0)
+            throw new InvalidArgumentException();
```

Comment:
When throwing an exception,  a message should always be provided in order to inform the user why the exception was thrown without needing to look at the source code